TCAT/Cells/RouteTableViewCell.swift

```diff
@@ -14,221 +14,314 @@ protocol TravelDistanceDelegate: NSObjectProtocol {
 
         routeDiagram.removeFromSuperview()
         cellSeperator.removeFromSuperview()
         bottomBorder.removeFromSuperview()
+        
+        // make live elements invisible
+        setLiveElements(withStartTime: nil, withDelay: nil)
     }
-    
+
     // MARK: Set Data
+
+    func setData(_ route: Route){
+        let firstDepartDirection = route.getFirstDepartDirection()
         
-    func setRouteData() {
-        
-        guard let departureTime = route?.departureTime,
-              let arrivalTime = route?.arrivalTime,
-              let isWalkingRoute = route?.isWalkingRoute(),
-              let directions = route?.directions,
-              let travelDistance = route?.travelDistance
-              else{
-                print("RouteTableViewCell route object does not have the data needed to fill in the cell")
-                return
-              }
+        setLiveElements(withFirstDepartDirection: firstDepartDirection)
         
-        setTravelTime(withDepartureTime: departureTime, withArrivalTime: arrivalTime)
-        setDepartureTime(withTime: departureTime, isWalkingRoute: isWalkingRoute)
+        let isWalkingRoute = route.isWalkingRoute()
         
-        routeDiagram.setRouteData(fromDirections: directions, fromTravelDistance: travelDistance)
+        if let firstDepartDirection = firstDepartDirection, let lastArriveDirection = route.getLastArriveDirection(){
+            setTravelTime(withDepartureTime: firstDepartDirection.startTime, withArrivalTime: lastArriveDirection.endTime)
+        }
+        else {
+            setTravelTime(withDepartureTime: route.departureTime, withArrivalTime: route.arrivalTime)
+        }
         
+        if let firstDepartDirection = firstDepartDirection {
+            setDepartureTime(withTime: firstDepartDirection.startTime, withWalkingRoute: isWalkingRoute)
+        }
+        else {
+            setDepartureTime(withTime: route.departureTime, withWalkingRoute: isWalkingRoute)
+        }
+
+        routeDiagram.setData(withDirections: route.directions, withTravelDistance: route.travelDistance, withWalkingRoute: isWalkingRoute)
     }
     
+    func setLiveElements(withFirstDepartDirection firstDepartDirection: Direction?) {
+        if let direction = firstDepartDirection {
+            Network.getBusLocations(routeID: String(direction.routeNumber), tripID: direction.tripID, stopID: direction.startLocation.id)
+                .perform(withSuccess: { (result) in
+                    if let busLocation = result.busLocation {
+                        
+                        switch busLocation.dataType {
+                            
+                        case .validData:
+                            self.setLiveElements(withStartTime: direction.startTime, withDelay: busLocation.delay)
+                            
+                        default:
+                           self.setLiveElements(withStartTime: nil, withDelay: nil)
```

Comment:
a lot of similar code for many cases. Could this be combined, or put in a completion function to not duplicate code?

Reply:
Maybe an enum for each scenario? I wanted to adjust the circle dot view, and realizes the three parameters were only being used in specific ways. It would maybe be a bit more understandable to have the cases be an enum

Reply:
Okay for me all I would need is a color parameter and size b/c looking at the Zeplin your RSS feed icon is 8.5 by 8.5 and mine's is 10.5 by 10.5. So we could have an enum like this:
````
enum RSSSize: Int {
    case small = 8.5
    case medium = 10.5
```

And then the init for the RSSFeedIcon could look like this:
```
init(size: RSSSize, color: UIColor) {
```